server/ClearableFileCache.ts

```diff
@@ -0,0 +1,242 @@
+import path from 'path';
+import LRUCache from 'lru-cache';
+import {CacheHandler, CacheHandlerContext, CacheHandlerValue} from 'next/dist/server/lib/incremental-cache';
+import {CacheFs} from 'next/dist/shared/lib/utils';
+import {IncrementalCacheValue} from 'next/dist/server/response-cache';
+
+type FileSystemCacheContext = Omit<
+    CacheHandlerContext,
+    'fs' | 'serverDistDir'
+> & {
+  fs: CacheFs
+  serverDistDir: string
+};
+
+let memoryCache: LRUCache<string, CacheHandlerValue> | undefined;
+
+export default class ClearableFileCache implements CacheHandler {
+  private fs: FileSystemCacheContext['fs'];
+  private flushToDisk?: FileSystemCacheContext['flushToDisk'];
+  private serverDistDir: FileSystemCacheContext['serverDistDir'];
+  private appDir: boolean;
+
+  constructor(ctx: FileSystemCacheContext) {
+    this.fs = ctx.fs;
+    this.flushToDisk = ctx.flushToDisk;
+    this.serverDistDir = ctx.serverDistDir;
+    this.appDir = !!ctx._appDir;
+
+    if (ctx.maxMemoryCacheSize && !memoryCache) {
+      const config = {
+        max: ctx.maxMemoryCacheSize,
+        maxSize: 10 * ctx.maxMemoryCacheSize, // had to be used sizeCalculation function
+        sizeCalculation({value}: { value: IncrementalCacheValue | null }) {
+          if (!value) {
+            return 25;
+          } else if (value.kind === 'REDIRECT') {
```

Comment:
If the `value.kind` is not a string, but rather a string literal union type, it may be better to move `value.kind` checks to the separate function with `switch` without `default`: the typescript will throw "not all paths return a value" error if more possible values will be added in the future in the library, so you don't forget to handle them all. Otherwise, new cases will go straigh to line 47.

Reply:
It's just a suggestion, so you might ignore it.

Reply:
This file is a copy of the nextjs' implementation with 2 methods added, `delete` and `clear` so I would like to keep it untouched as much as possible ;)